src/core/Transcriber.py

```diff
@@ -1,90 +1,241 @@
+            if not result_queue.empty():
+                result_data = result_queue.get()
+                
+                if result_data["success"]:
+                    result = result_data["result"]
+                    
+                    # Store results
+                    self.whisper_result = result
+                    self.ts_api.database.update_job_status(
+                        self.module_entry.uid,
+                        JobStatus.COMPLETED,
+                        whisper_result=result,
+                        completed_at=time.time()
+                    )
+                    
+                    # Clean up audio file
+                    if os.path.exists(self.file_path):
+                        os.remove(self.file_path)
+                        logging.debug(f"Removed audio file for job {self.module_entry.uid}")
+                    
+                    logging.info(f"Successfully completed job {self.module_entry.uid}")
+                else:
+                    raise Exception(result_data["error"])
+            else:
+                # Process exited without result
+                raise Exception("Transcription process exited without result")
+            
         except Exception as e:
-            logging.error(e)
-            self.ts_api.database.change_job_entry(self.module_entry.uid,
-                                                  "status", 4)  # Failed
-            self.ts_api.unregister_job(self.module_entry)
+            logging.error(f"Error transcribing job {self.module_entry.uid}: {e}")
+            self.ts_api.database.update_job_status(
+                self.module_entry.uid,
+                JobStatus.FAILED,
+                error_message=str(e),
+                completed_at=time.time()
+            )
+            
+            # Clean up audio file on error
+            if os.path.exists(self.file_path):
+                try:
+                    os.remove(self.file_path)
+                except:
+                    pass
+            
+            # Clean up process if still running
+            if self.transcription_process and self.transcription_process.is_alive():
+                self.transcription_process.terminate()
+                self.transcription_process.join(timeout=2)
+    
+    def _handle_cancellation(self):
+        """Handle job cancellation cleanly"""
+        logging.info(f"Handling cancellation for job {self.module_entry.uid}")
+        
+        # Update status
+        self.ts_api.database.update_job_status(
+            self.module_entry.uid,
+            JobStatus.CANCELED,
+            completed_at=time.time()
+        )
+        
+        # Clean up audio file
+        if os.path.exists(self.file_path):
+            try:
+                os.remove(self.file_path)
+                logging.debug(f"Removed audio file for canceled job {self.module_entry.uid}")
+            except Exception as e:
+                logging.error(f"Failed to remove audio file: {e}")
```

Comment:
Nit: would it be better to specify the file that we cancel?
```suggestion
                logging.error(f"Failed to remove audio file for canceled job {self.module_entry.uid}: {e}")
```